internal/v4/auth.go

```diff
@@ -110,17 +110,26 @@ func (h *Handler) authorizeWithPerms(req *http.Request, read, write []string) er
 	return h.authorize(req, acl)
 }
 
-const usernameAttr = "username"
+const (
+	usernameAttr = "username"
+	groupsAttr   = "groups"
+)
 
 func (h *Handler) checkACLMembership(attrs map[string]string, acl []string) error {
 	user := attrs[usernameAttr]
 	if user == "" {
 		return errgo.New("no username declared")
 	}
+	groups := strings.Fields(attrs[groupsAttr])
 	for _, name := range acl {
```

Comment:
+1 (although for practical purposes it will probably almost always
be slower to do that, as most users aren't a member of many groups).